src/rgw/driver/rados/rgw_zone.cc

```diff
@@ -1199,6 +1199,16 @@ int SiteConfig::load(const DoutPrefixProvider* dpp, optional_yield y,
   } else if (realm) {
     // load the realm's default zone
     r = cfgstore->read_default_zone(dpp, y, realm->id, zone_params, nullptr);
+    if (r == -ENOENT) {
+      ldpp_dout(dpp, 0) << "No rgw_zone configured, and the selected realm \""
+          << realm->name << "\" does not have a default zone." << dendl;
+      if (realm_name.empty()) {
```

Comment:
you probably meant not `!realm_name.empty()` ?

Reply:
from the block just above:
```c++
  // try to load a realm
  realm.emplace();
  std::string realm_name = dpp->get_cct()->_conf->rgw_realm;
  if (!realm_name.empty()) {
    r = cfgstore->read_realm_by_name(dpp, y, realm_name, *realm, nullptr);
  } else {
    r = cfgstore->read_default_realm(dpp, y, *realm, nullptr);
```

when `!realm_name.empty()`, we try to load that realm by name. it's the `realm_name.empty()` case that consults the 'default realm' that this new error message refers to